src/pykeen/nn/node_piece/anchor_search.py

```diff
@@ -217,6 +219,95 @@ def __call__(self, edge_index: numpy.ndarray, anchors: numpy.ndarray, k: int) ->
         return self.select(pool=pool, k=k)
 
 
+class PersonalizedPageRankAnchorSearcher(AnchorSearcher):
+    """Select closest anchors as the nodes with the largest personalized page rank."""
+
+    def __init__(self, batch_size: int = 1, use_tqdm: bool = False, page_rank_kwargs: OptionalKwargs = None):
+        """
+        Initialize the searcher.
+
+        :param batch_size:
+            the batch size to use.
+        :param use_tqdm:
+            whether to use tqdm
+        :param page_rank_kwargs:
+            keyword-based parameters used for :func:`page_rank`. Must not include `edge_index`, or `x0`.
+        """
+        self.batch_size = batch_size
+        self.page_rank_kwargs = page_rank_kwargs or {}
+        self.use_tqdm = use_tqdm
+
+    def extra_repr(self) -> Iterable[str]:  # noqa: D102
+        yield f"batch_size={self.batch_size}"
+        yield f"use_tqdm={self.use_tqdm}"
+        yield f"page_rank_kwargs={self.page_rank_kwargs}"
+
+    def precalculate_anchor_ppr(self, edge_index: numpy.ndarray, anchors: numpy.ndarray) -> numpy.ndarray:
+        """
+        Sort anchors nodes by PPR values from each node.
+
+        :param edge_index: shape: (2, m)
+            the edge index.
+        :param anchors: shape: `(num_anchors,)`
+            the anchor IDs.
+
+        :return: shape: `(num_entities, num_anchors)`
+            the PPR values for each anchor
+        """
+        return numpy.concatenate(
+            [
+                numpy.argsort(ppr_batch, axis=-1)
+                for ppr_batch in self._iter_ppr(
+                    edge_index=edge_index,
+                    anchors=anchors,
+                )
+            ]
+        )[:, ::-1]
+
+    def __call__(self, edge_index: numpy.ndarray, anchors: numpy.ndarray, k: int) -> numpy.ndarray:  # noqa: D102
+        n = edge_index.max().item() + 1
+        result = numpy.full(shape=(n, k), fill_value=-1)
+        i = 0
+        for batch_ppr in self._iter_ppr(edge_index=edge_index, anchors=anchors):
+            batch_size = batch_ppr.shape[1]
```

Comment:
this line is apparently a type - should be `batch_ppr.shape[0]` according to the docstring of `_iter_ppr()` that returns a shape `(batch_size,  num_anchors)` 

so if number of anchors is, say, 500, it would update 500 rows in the result matrix, but should update only one :)

Reply:
cf. f6fdc3d011d6d4e1ade6d6bdaa4c16065f30b04d